sheets-import/util.gs

```diff
@@ -0,0 +1,403 @@
+function randomString(l) {
+  l = l || 14;
+  return Math.random().toString(36).substring(2, l + 2);
+}
+
+/**
+ * Builds and returns the API authorization URL from the service object.
+ * @return {String} the API authorization URL.
+ */
+function getAuthorizationUrl() {
+  return getService().getAuthorizationUrl();
+}
+
+/**
+ * Resets the API service, forcing re-authorization before
+ * additional authorization-required API calls can be made.
+ */
+function signout() {
+  getService().reset();
+}
+
+/**
+ * Sheets-specific utility. Find a sheet within a spreadsheet with
+ * the given id. If not present, return null.
+ * @param {Object} ss a Spreadsheet object.
+ * @param {Number} sheetId a Sheet id.
+ * @return {Object} a Sheet object, or null if not found.
+ */
+function getSheetById(ss, sheetId) {
+  if (sheetId === null) {
+    return null;
+  }
+  var sheets = ss.getSheets();
+  for (var i = 0; i < sheets.length; i++) {
+    if (sheets[i].getSheetId() === sheetId) {
+      return sheets[i];
+    }
+  }
+  return null;
+}
+
+/**
+ * Sheets-specific utility. Given a base title for a sheet, check
+ * for that it is unique in the spreadsheet. If not, find an integer
+ * suffix to append to it to make it unique and return. This function
+ * is used to avoid name collisions while adding or renaming sheets
+ * automatically.
+ * @param {Object} spreadsheet a Spreadsheet.
+ * @param {String} baseName the initial suggested title for a sheet.
+ * @return {String} a unique title for the sheet, based on the
+ *     given base title.
+ */
+function getUniqueSheetName(spreadsheet, baseName) {
+  var sheetName = baseName;
+  var i = 2;
+  while (spreadsheet.getSheetByName(sheetName) !== null) {
+      sheetName = baseName + ' ' + i++;
+  }
+  return sheetName;
+}
+
+/**
+ * Sheets-specific utility. Given a spreadsheet and a triggerId string,
+ * return the user trigger that corresponds to that ID. Returns null
+ * if no such trigger exists.
+ * @param {Spreadsheet} spreadsheet container of the user triggers.
+ * @param {String} triggerId trigger ID string.
+ * @return {Trigger} corresponding user trigger, or null if not found.
+ */
+function getUserTriggerById(spreadsheet, triggerId) {
+  var triggers = ScriptApp.getUserTriggers(spreadsheet);
+  for (var i = 0; i < triggers.length; i++) {
+    if (triggers[i].getUniqueId() == triggerId) {
+      return triggers[i];
+    }
+  }
+  return null;
+}
+
+/**
+ * Sheets-specific utility. Given a String sheet id, activate that
+ * sheet if it exists.
+ * @param {String} sheetId the sheet ID.
+ */
+function activateById(sheetId) {
+  var ss = SpreadsheetApp.getActiveSpreadsheet();
+  var sheet = getSheetById(ss, parseInt(sheetId));
+  if (sheet !== null) {
+    sheet.activate();
+  }
+}
+
+/**
+ * Returns true if the current user is allowed to edit the
+ * report associated with the given config; returns
+ * false otherwise.
+ * @param {Object} config a report configuration.
+ */
+function canEditReport(config) {
```

Comment:
A function defined in a "utilities" file carries the connotation that the function is generic and can be used as is in other projects.  You've places several functions here that are specific to report creation/manipulation -- they aren't really utilities.  Consider breaking them out into a separate file.